spdx-utils/src/main/kotlin/Extensions.kt

```diff
@@ -52,7 +52,7 @@ infix fun SpdxLicense.or(other: SpdxExpression) =
  * Create an [SpdxExpression] by concatenating [this][SpdxLicense] and [exception] using [SpdxOperator.WITH].
  */
 infix fun SpdxLicense.with(exception: SpdxLicenseException) =
-    SpdxCompoundExpression(toExpression(), SpdxOperator.WITH, exception.toExpression())
+    SpdxLicenseWithExceptionExpression(toExpression(), exception.toExpression())
```

Comment:
Did you consider making the exception `nullable`? context: I thought about which parts of the code would want to handle the licenses at which level on this abstraction hierarchy - and to me `SpdxLicenseWithExceptionExpression` seemed a good fit to almost everywhere if exception was `nullable`. Even the notice reporter could work on this type (and just ignore the exception for now) as it already does. What do you think?

Below `decompose()` would then become quite simple, e.g. always return `SpdxLicenseWithExceptionExpression`...

Reply:
As discussed offline, I'll keep it as is, because with sealed classes the different cases can easily be handled with a `when` expression checking for the type, instead of e.g. checking for `exception == null`.